### ISSUE TITLE
Handle SIGINT (ctrl + c) in migrate CLI

### ISSUE THREAD

**Issue by haochuan** [rehearsal-js/rehearsal-js#441]

**Comment by haochuan** [comment]
@lynchbomb More info:
All the other `listr2` tasks inside CLI handles `SIGINT` well, the only probably is when we calling `migrate` package, the CLI would still continue but it could not be killed by `ctrl + c`.

What I found so far is inside this for loop (https://github.com/rehearsal-js/rehearsal-js/blob/master/packages/migrate/src/migrate.ts#L97-L106), there is a spawned process which doesn't access `SIGINT`.

**Comment by chadhietala** [comment]
@lynchbomb I swore you mentioned that this was done?

**Comment by lynchbomb** [comment]
We can successfully SIGINT from the `git mv` tasks after moving away from execa. However there is another child process ignoring SIGINT which stems from `DiagnosticFixPlugin`. I am currently debugging this - im pretty sure its coming from the ts language service.

**Comment by lynchbomb** [comment]
Depends on #630

**Comment by lynchbomb** [comment]
with interactive mode as default this is only ever an issue if it were run manually with `--ci` as in CI the process will exit on error and wouldn't have an instance where we would have to ctrl + c

**Comment by haochuan** [comment]
@lynchbomb FYI
Check this again with #886, still not able to ctrl + c to stop the CLI whiling processing files

**Comment by lynchbomb** [comment]
Yeah its because we still have the blocking `while (diagnostics.length > 0) {` while loops in the plugins
